src/VisualStudio/Core/Def/Implementation/Options/FeatureFlagPersisterProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
+{
+    [Export(typeof(IOptionPersisterProvider))]
+    internal sealed class FeatureFlagPersisterProvider : IOptionPersisterProvider
+    {
+        private readonly IAsyncServiceProvider _serviceProvider;
+        private FeatureFlagPersister? _lazyPersister;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FeatureFlagPersisterProvider(SVsServiceProvider serviceProvider)
+        {
+            _serviceProvider = (IAsyncServiceProvider)serviceProvider;
+        }
+
+        public async ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
+        {
+            if (_lazyPersister != null)
+            {
+                return _lazyPersister;
+            }
+
+            IVsFeatureFlags? service;
+            try
+            {
+                service = (IVsFeatureFlags?)await _serviceProvider.GetServiceAsync(typeof(SVsFeatureFlags)).ConfigureAwait(false);
```

Comment:
safe to make this cast on a BG thread? (in general IVs casts are not).  but if this is known to be free threaded/non-STA, then fine.

Reply:
Previously this was run on background thread via JTF.Run:
https://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Experimentation/VisualStudioExperimentationService.cs,43